class-wc-calypso-bridge.php

```diff
@@ -133,24 +132,32 @@ public function load_ecommerce_plan_ui() {
 	}
 
 	function add_extension_register_script() {
-		if ( ! class_exists( 'Automattic\WooCommerce\Admin\Loader' ) || ! \Automattic\WooCommerce\Admin\Loader::is_admin_or_embed_page() ) {
-			return;
-		}
+		$is_woo_page = class_exists( 'Automattic\WooCommerce\Admin\Loader' ) 
+			&& \Automattic\WooCommerce\Admin\Loader::is_admin_or_embed_page() 
+			? true 
+			: false;
 		
 		$script_path       = '/build/index.js';
 		$script_asset_path = dirname( __FILE__ ) . '/build/index.asset.php';
 		$script_asset      = file_exists( $script_asset_path )
 			? require( $script_asset_path )
 			: array( 'dependencies' => array(), 'version' => filemtime( $script_path ) );
 		$script_url = plugins_url( $script_path, __FILE__ );
-	
+
 		wp_register_script(
 			'wc-calypso-bridge',
 			$script_url,
 			$script_asset['dependencies'],
 			$script_asset['version'],
 			true
 		);
+
+		wp_add_inline_script( 'wc-calypso-bridge', 'window.wcCalypsoBridge = ' . wp_json_encode(
+			array(
+				'isWooPage' => $is_woo_page,
+				'homeUrl'   => get_home_url()
+			)
+		), 'before' );
```

Comment:
There's a system for this kind of thing that WC Admin uses and Blocks as well to retrieve values with `@woocommerce/wc-admin-settings`. See it used in the homescreen:

https://github.com/woocommerce/woocommerce-admin/blob/ec8471dcadc2a91f1e972c84c2d6970e5c84900a/src/Features/Homescreen.php#L53

The intension was for extensions to also use it. I'm rather torn on whether or not we should use it here. Its the "right" way to do things, but that system is less than ideal because the `getSetting` method was a mistake because it manipulates the object on the window. We mitigated that to a certain extent with our data stores' hydration. I'm feeling like its probably best to avoid it, but just wanted to share some background in case you had any feelings either way.

Reply:
Thanks for this insight @psealock , good to see the context here. It looks like that filter ultimately enables you to add properties to `wcSettings` on the `window`, although that's only available when you're on a Woo page. In this case I'd ideally have access when on _any_ page (including WP Admin). 

Also, it looks like this filter is applied here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/3c55aac1b599a5e4a8d2bf8ec56e89f6c56b3ad5/src/Assets/AssetDataRegistry.php#L130

But that also includes this comment, which makes it sound like we shouldn't be using it?
```

Reply:
Ah thats right, we should be using `Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry::class` but the old way `woocommerce_shared_settings` is still supported. 

>only available when you're on a Woo page

This would be a blocker then. Thanks for looking into it!